precompiles/bank/integration_test.go

```diff
@@ -0,0 +1,182 @@
+package bank_test
+
+import (
+	"math/big"
+
+	evmosutiltx "github.com/evmos/evmos/v15/testutil/tx"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v15/precompiles/bank"
+
+	"github.com/evmos/evmos/v15/precompiles/testutil"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/keyring"
+	//nolint:revive // dot imports are fine for Ginkgo
+	. "github.com/onsi/ginkgo/v2"
+	//nolint:revive // dot imports are fine for Ginkgo
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Bank Extension -", func() {
+	var (
+		// BankCallerContractAddr         common.Address
+		err    error
+		sender keyring.Key
+		amount *big.Int
+
+		// contractData is a helper struct to hold the addresses and ABIs for the
+		// different contract instances that are subject to testing here.
+		contractData ContractData
+		passCheck    testutil.LogCheckArgs
+	)
+
+	BeforeEach(func() {
+		s.SetupTest()
+
+		// Default sender and amount
+		sender = s.keyring.GetKey(0)
+		amount = big.NewInt(1e18)
+
+		contractData = ContractData{
+			ownerPriv:      sender.Priv,
+			precompileAddr: s.precompile.Address(),
+			precompileABI:  s.precompile.ABI,
+		}
+
+		passCheck = testutil.LogCheckArgs{}.WithExpPass(true)
+
+		err = s.network.NextBlock()
+		Expect(err).ToNot(HaveOccurred(), "failed to advance block")
+	})
+
+	Context("Direct precompile queries", func() {
+		Context("balances query", func() {
+			It("should return the correct balance", func() {
+				balanceBefore, err := s.grpcHandler.GetBalance(sender.AccAddr, "xmpl")
+				Expect(err).ToNot(HaveOccurred(), "failed to get balance")
+				Expect(balanceBefore.Balance.Amount).To(Equal(sdk.NewInt(0)))
+				Expect(balanceBefore.Balance.Denom).To(Equal("xmpl"))
```

Comment:
would be better to define a constant `tokenDenom := "xmpl"` and reuse throughout the tests